src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,2449 @@
+        Dim i = New I()
+                    ~
+                                        </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub CoClassWithRequiredMembers_AllSet()
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers("
+using System;
+using System.Runtime.InteropServices;
+
+[ComImport, Guid(""00020810-0000-0000-C000-000000000046"")]
+[CoClass(typeof(C))]
+public interface I
+{
+    public int P { get; set; }
+}
+
+public class C : I
+{
+    public required int P { get; set; }
+}
+")
+            Dim comp = CreateCompilation("
+Module M
+    Sub Main()
+        Dim i = New I() With { .P = 1 }
+    End Sub
+End Module", {cComp.EmitToImageReference()})
+
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public Overloads Property P As Integer' must be set in the object initializer or attribute arguments.
+        Dim i = New I() With { .P = 1 }
+                    ~
+                                        </expected>)
+        End Sub
+
+        Public Function GetAttributeDefinition(hasSetsRequiredMembers As Boolean) As String
+            Return $"
+using System;
+public class AttrAttribute : Attribute
+{{
+    {If(hasSetsRequiredMembers, "[SetsRequiredMembers]", "")}
+    public AttrAttribute() {{}}
+
+    public required int P {{ get; set; }}
+}}
+"
+        End Function
+
+        <Fact>
+        Public Sub RequiredMemberInAttribute_NoneSet()
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetAttributeDefinition(hasSetsRequiredMembers:=False))
+
+            Dim vbCode = $"
+<Attr>
+Class C
+End Class"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertTheseDiagnostics(<expected><![CDATA[
+BC37321: Required member 'Public Overloads Property P As Integer' must be set in the object initializer or attribute arguments.
+<Attr>
+ ~~~~
+]]></expected>)
+        End Sub
+
+        <Fact>
+        Public Sub RequiredMemberInAttribute_AllSet()
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetAttributeDefinition(hasSetsRequiredMembers:=False))
+
+            Dim vbCode = $"
+<Attr(P:=1)>
+Class C
+End Class"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub PublicAPITests()
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetCDefinition(hasSetsRequiredMembers:=False))
+
+            Dim vbComp = CreateCompilation("", {cComp.EmitToImageReference()})
+
+            Dim c = vbComp.GetTypeByMetadataName("C")
+            Assert.False(c.HasRequiredMembersError)
+            Dim prop = c.GetMember(Of PropertySymbol)("Prop")
+            Dim field = c.GetMember(Of FieldSymbol)("Field")
+
+            AssertEx.Equal(Of Symbol)({field, prop}, From kvp In c.AllRequiredMembers
```

Comment:
nit: For tests that verify new symbol APIs like `AllRequiredMembers`, it would also be good to make the type of the symbol apparent in the test (for example, `RetargetingPropertySymbol`, ...). 
That makes it easy to verify which symbols were covered